poetry/utils/env.py

```diff
@@ -834,13 +834,16 @@ def remove_venv(cls, path):  # type: (Union[Path,str]) -> None
             return
         except OSError as e:
             # Continue only if e.errno == 16
-            if e.errno != 16:  # ERRNO 16: Device or resource busy
+            if (
+                e.errno != 16 and e.errno != 20
+            ):  # ERRNO 16: Device or resource busy, # ERRNO 20: Not a directory
                 raise e
 
         # Delete all files and folders but the toplevel one. This is because sometimes
-        # the venv folder is mounted by the OS, such as in a docker volume. In such
+        # the venv folder is mounted or symlinked by the OS, such as in a docker volume, codebuild, etc. In such
         # cases, an attempt to delete the folder itself will result in an `OSError`.
         # See https://github.com/python-poetry/poetry/pull/2064
+        # See https://bugs.python.org/issue1669
```

Comment:
I don't think this issue is relevant -- it describes a bug that is long-resolved, and not the behavior of throwing an exception when called on a symlink.